src/status_im/ui/screens/wallet/events.cljs

```diff
@@ -18,107 +19,114 @@
          (on-error err))))
     (on-error "web3 or account-id not available")))
 
+(defn assoc-error-message [db error-type err]
+  (assoc-in db [:wallet :errors error-type] (or (when err (str err))
+                                                :unknown-error)))
+
+(defn clear-error-message [db error-type]
+  (update-in db [:wallet :errors] dissoc error-type))
+
 ;; FX
 
 (reg-fx
   :get-balance
   (fn [{:keys [web3 account-id success-event error-event]}]
-    (get-balance
-     {:web3           web3
-      :account-id     account-id
-      :on-success     #(dispatch [success-event %])
-      :on-error       #(dispatch [error-event %])})))
+    (get-balance {:web3           web3
+                  :account-id     account-id
+                  :on-success     #(dispatch [success-event %])
+                  :on-error       #(dispatch [error-event %])})))
 
 (reg-fx
   :get-transactions
-  (fn [{:keys [network account-id]}]
+  (fn [{:keys [network account-id success-event error-event]}]
     (transactions/get-transactions network
                                    account-id
-                                   #(dispatch [:update-transactions-succes %])
-                                   #(dispatch [:update-transactions-fail %]))))
+                                   #(dispatch [success-event %])
+                                   #(dispatch [error-event %]))))
 
 (reg-fx
   :get-prices
   (fn [{:keys [from to success-event error-event]}]
-    (prices/get-prices
-     from
-     to
-     #(dispatch [success-event %])
-     #(dispatch [error-event %]))))
+    (prices/get-prices from
+                       to
+                       #(dispatch [success-event %])
+                       #(dispatch [error-event %]))))
 
 ;; Handlers
 
-;; TODO(oskarth): At some point we want to get list of relevant assets to get prices for
 (handlers/register-handler-fx
-  :load-prices
-  (fn [{{:keys [wallet] :as db} :db} [_ a]]
-    {:get-prices  {:from          "ETH"
-                   :to            "USD"
-                   :success-event :update-prices
-                   :error-event   :update-prices-fail}
-     :db (assoc db :prices-loading? true)}))
-
-(handlers/register-handler-fx
-  :refresh-wallet
+  :update-wallet
   (fn [{{:keys [web3 accounts/current-account-id network] :as db} :db} [_ a]]
     {:get-balance {:web3          web3
                    :account-id    current-account-id
-                   :success-event :update-balance
+                   :success-event :update-balance-success
                    :error-event   :update-balance-fail}
-     :dispatch    [:load-prices]
+     :dispatch    [:update-prices]
```

Comment:
I made a change tell me if it was you meant

Reply:
Not really, what I meant was to extract `:update-prices` handler into named function, like this:
```clj
(defn update-prices [db]
  {:get-prices  {:from "ETH"
                 :to   "USD"
   :success-event :update-prices-success
   :error-event   :update-prices-fail}
   :db (assoc db :prices-loading? true)}))

;; This handler registration is only necessary when `:update-prices` ever gets called as result of user action
(handlers/register-handler-fx
  :update-prices
  (fn [{:keys [db]} _]
    (update-prices db)))
```
Then in the `:update-wallet` handlers, instead of indicating dispatch, you can do:
```clj
(merge {:web3          web3
        :account-id    current-account-id
        :success-event :update-balance-success
        :error-event   :update-balance-fail}
       (update-prices db)))
````